File: c/perfect-numbers/perfect_numbers.c
#include "perfect_numbers.h"

kind classify_number(int number)
{
    if ( number <= 0)
    {
        return ERROR;
    }

    int sum = 0;

    for(int i = 1; i < number; i++)
    {
        if (number % i == 0)
        {
            sum = sum + i;
        }
    }

    if(number - sum < 0)
    {
        return ABUNDANT_NUMBER;
    }
    if(number - sum > 0)
    {
        return DEFICIENT_NUMBER;
    }
    return PERFECT_NUMBER;
}

File: csharp/perfect-numbers/PerfectNumbers.cs
public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number < 1)
        {
            throw new ArgumentOutOfRangeException();
        }
        int sum = 0;
        for (int i = 1; i <= number / 2; i++)
        {
            if (number % i == 0)
            {
                sum += i;
            }
        }
        if (sum > number) return Classification.Abundant;
        if (sum < number) return Classification.Deficient;
        return Classification.Perfect;
    }
}
